server/src/main/java/io/crate/replication/logical/action/CreateSubscriptionRequest.java

```diff
@@ -55,7 +56,7 @@ public CreateSubscriptionRequest(StreamInput in) throws IOException {
         this.owner = in.readString();
         this.name = in.readString();
         connectionInfo = new ConnectionInfo(in);
-        this.publications = List.of(in.readStringArray());
+        this.publications = in.readList(StreamInput::readString);
```

Comment:
List.of creates ImmutableList making `if (list.removeAll(droppedPublications))` fail